### PR TITLE
MBS-8134: Allow amending new entities of any type

### PR DESCRIPTION
**Pull request by yvanzo**
Implement [MBS-8134](https://tickets.metabrainz.org/browse/MBS-8134): Allow to _amend_ new entities created by the same editor less than an hour ago.

- Parameterize allow-amending Edit role to support any entity type beyond Release.
- Add tests based on the example of amending medium format for a release.
- Extend to Artist, Event, Label, Place, Recording, Release Group, Series, and Work.

### CODE REVIEW

**Comment by yvanzo** [comment]
@brainzbot, add to whitelist.

**Comment by mwiencek** [comment]
@brainzbot, add to whitelist.

File: lib/MusicBrainz/Server/Edit/Role/AllowAmending.pm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+=cut
+
+    method 'can_amend' => sub {
+        my ($self, $amended_entity_id) = @_;
+
+        my $add_entity_edit = $self->c->sql->select_single_value("
+            SELECT id FROM edit
+              JOIN edit_$entity_type ON edit.id = edit_$entity_type.edit
+             WHERE edit_$entity_type.$entity_type = $amended_entity_id
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Edit/Role/AllowAmending.pm]: I'd prefer using `?` placeholders for the values like `AllowAmendingRelease` did (though I think it's probably fine for immutable constants defined in the code, like `$create_edit_type`).

File: lib/MusicBrainz/Server/Edit/Series/Edit.pm [hunk 2]
```diff
@@ -82,6 +89,8 @@ sub build_display_data {
 around allow_auto_edit => sub {
     my ($orig, $self, @args) = @_;
 
+    return 1 if $self->can_amend($self->series_id);
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Edit/Series/Edit.pm]: Will that allow people to change the series type if they've already added items of a different type?

Comment (yvanzo) [inline lib/MusicBrainz/Server/Edit/Series/Edit.pm]: No, it doesn’t, it triggers an ISE as it always did. That should probably be fixed apart.

Comment (mwiencek) [inline lib/MusicBrainz/Server/Edit/Series/Edit.pm]: Yeah, sorry, I realized after I wrote that that whether or not it's an auto-edit shouldn't affect that.

**Comment by mwiencek** [comment]
This looks great. :shipit:

**Pull request merged**